wagtail/admin/views/pages/search.py

```diff
@@ -41,6 +42,11 @@ def page_filter_search(q, pages, all_pages=None, ordering=None):
 @user_passes_test(user_has_any_page_permission)
 def search(request):
     pages = all_pages = Page.objects.all().prefetch_related('content_type').specific()
+    show_locale_labels = getattr(settings, 'WAGTAIL_I18N_ENABLED', False)
+    if show_locale_labels:
+        pages = pages.select_related('locale')
+        all_pages = all_pages.select_related('locale')
```

Comment:
`all_pages` is only used for displaying the count of results, so the `select_related` can be omitted here I think.

Reply:
I somehow missed that in my search. Sorted in ecfffaa
imho, we should only pass the count to the search results template to avoid further confusion, but that is for another PR